### PR TITLE
Set up application structure for API

### PR DESCRIPTION
**Pull request by Carltesio**
# [Set up application structure for API](https://www.pivotaltracker.com/n/projects/2438205)

## Changes proposed in this pull request:
* Created new set up branch
* Installed rails scaffold within the cloned folder from the upstream
* Modified config/application.rb to allow get, post, put, delete requests from any origin and any sourc
* modified gemfile and added the rack-cors gem

## Packages/Gems added
rack-cors gem to allow external clients to access our application.

### CODE REVIEW

File: app/channels/application_cable/channel.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
```
Comment (hmvitous) [inline app/channels/application_cable/channel.rb:4]: I dont remember this in past projects
Do we need this?

File: app/views/layouts/mailer.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (kaylawoodbury) [inline app/views/layouts/mailer.html.erb:1]: Do we need this file? I don't think we are working with views on the API

File: app/views/layouts/mailer.text.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= yield %>
```
Comment (kaylawoodbury) [inline app/views/layouts/mailer.text.erb:1]: Do we need this file? I don't think we are working with views on the API

File: app/views/layouts/mailer.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
+    </style>
+  </head>
+
+  <body>
+    <%= yield %>
+  </body>
+</html>
```
Comment (hmvitous) [inline app/views/layouts/mailer.html.erb:13]: We may not need the views since it is an api.

File: app/views/layouts/mailer.text.erb (hunk 3 above)
Comment (hmvitous) [inline app/views/layouts/mailer.text.erb:1]: We may not need the views since it is an api.

File: bin/bundle [hunk 5]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+# This file was generated by Bundler.
+#
+# The application 'bundle' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
```
Comment (kaylawoodbury) [inline bin/bundle]: I think we can remove these commented out lines

Comment (hmvitous) [inline bin/bundle]: Delete commented text

File: bin/setup [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+FileUtils.chdir APP_ROOT do
+  # This script is a way to setup or update your development environment automatically.
+  # This script is idempotent, so that you can run it at anytime and get an expectable outcome.
+  # Add necessary setup steps to this file.
+
```
Comment (kaylawoodbury) [inline bin/setup:10]: I think we can remove the commented out lines here, same with most of the files

File: bin/setup [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  puts "\n== Preparing database =="
+  system! 'bin/rails db:prepare'
+
+  puts "\n== Removing old logs and tempfiles =="
+  system! 'bin/rails log:clear tmp:clear'
+
+  puts "\n== Restarting application server =="
+  system! 'bin/rails restart'
+end
```
Comment (hmvitous) [inline bin/setup:23]: Delete commented text

File: config/application.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+require "action_mailer/railtie"
+require "action_mailbox/engine"
+require "action_text/engine"
+require "action_view/railtie"
+require "action_cable/engine"
+
+Bundler.require(*Rails.groups)
+
+module CocoktailApi
+  class Application < Rails::Application
```
Comment (hmvitous) [inline config/application.rb:17]: Typo CocoktailAPI should be CocktailAPI

File: db/seeds.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
```
Comment (kaylawoodbury) [inline db/seeds.rb]: again, think we can remove the commented out lines

**Review by hmvitous — COMMENTED** [review]
Check for commented out code.

**Review by kaylawoodbury — COMMENTED** [review]
There are a lot of commented out lines still that we could remove, but otherwise I am not sure if there are additional files/folders that should be removed. But we can remove them later if we realise they are irrelevant.

File: bin/bundle (hunk 5 above)
Comment (janko101) [reply bin/bundle]: did it

File: app/channels/application_cable/channel.rb (hunk 1 above)
Comment (janko101) [reply app/channels/application_cable/channel.rb:4]: I guess we do

File: app/views/layouts/mailer.html.erb (hunk 4 above)
Comment (janko101) [reply app/views/layouts/mailer.html.erb:13]: We choose to leave the views

File: app/views/layouts/mailer.text.erb (hunk 3 above)
Comment (janko101) [reply app/views/layouts/mailer.text.erb:1]: we choose to leave the views regardles

File: bin/setup (hunk 7 above)
Comment (janko101) [reply bin/setup:23]: deleted

File: config/application.rb (hunk 8 above)
Comment (janko101) [reply config/application.rb:17]: Fixed the typo everywhere. Thank you

File: app/views/layouts/mailer.html.erb (hunk 2 above)
Comment (janko101) [reply app/views/layouts/mailer.html.erb:1]: Not sure. We decided to leave it, until instructed otherwise.

File: bin/bundle (hunk 5 above)
Comment (janko101) [reply bin/bundle]: It's deleted

File: bin/setup (hunk 6 above)
Comment (janko101) [reply bin/setup:10]: Deleted

File: db/seeds.rb (hunk 9 above)
Comment (janko101) [reply db/seeds.rb]: Deleted

File: config/application.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+require "action_text/engine"
+require "action_view/railtie"
+require "action_cable/engine"
+
+Bundler.require(*Rails.groups)
+
+module CocktailApi
+  class Application < Rails::Application
+    config.load_defaults 6.0
+
```
Comment (oliverochman) [inline config/application.rb]: please turn off generators in this file

File: config/database.yml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+
+development:
+  <<: *default
+  database: cocoktail_api_development
```
Comment (oliverochman) [inline config/database.yml]: fix typos in this file

File: config/master.key [hunk 12]
```diff
@@ -0,0 +1 @@
+f8e0803e94fe9b7a796ab6a7ce6e2c4e
```
Comment (oliverochman) [inline config/master.key]: dont push up the master key!

put it in the gitignore and find a way to generate a new one

**Review by oliverochman — DISMISSED** [review]

File: config/master.key (hunk 12 above)
Comment (Carltesio) [reply config/master.key]: **Followed these instruction to regenerate it:**

No key regeneration feature at the moment. We have to do it manually.

1)Copy content of original credentials rails credentials:show somewhere temporarily.
2)Remove config/master.key and config/credentials.yml.enc
3)Run EDITOR=vim rails credentials:edit in the terminal: This command will create a new 4)master.key and credentials.yml.enc if they do not exist.
5)Paste the original credentials you copied (step 1) in the new credentials file (and save + 
   quit vim)
6)Add and Commit the file config/credentials.yml.enc

File: app/views/layouts/mailer.html.erb (hunk 4 above)
Comment (hmvitous) [reply app/views/layouts/mailer.html.erb:13]: Okay. No problem

File: app/views/layouts/mailer.text.erb (hunk 3 above)
Comment (hmvitous) [reply app/views/layouts/mailer.text.erb:1]: No problem

**Review by hmvitous — COMMENTED** [review]
It looks good to me. I feel a coach needs to double check because I do not know 100% what is necessary and what is not.

**Pull request merged**